tests/test_urlhaus.py

```diff
@@ -0,0 +1,469 @@
+from unittest.mock import patch, Mock
+import pytest
+import requests
+import json
+from modules.threat_intelligence.urlhaus import (
+    URLhaus,
+    ThreatLevel,
+    EvidenceType,
+    IDEACategory,
+    Direction,
+    URLHAUS_BASE_URL,
+)
+
+
+@pytest.mark.parametrize(
+    "to_lookup, expected_url, status_code",
+    [
+        # Testcase1: URL lookup
+        (
+            {"url": "https://example.com"},
+            f"{URLHAUS_BASE_URL}/url/",
+            200,
+        ),
+        # Testcase2: hash lookup
+        (
+            {"md5_hash": "a1b2c3d4"},
+            f"{URLHAUS_BASE_URL}/payload/",
+            200,
+        ),
+    ],
+)
+@patch("modules.threat_intelligence.urlhaus.requests.session")
+def test_make_urlhaus_request(
+    mock_response, mock_db, to_lookup, expected_url, status_code
+):
+    """Test successful requests to the make_urlhaus_request function."""
+    mock_response_instance = Mock()
+    mock_response.return_value = mock_response_instance
+    mock_response = Mock(status_code=status_code)
+    mock_response_instance.post.return_value = mock_response
+
+    urlhaus = URLhaus(mock_db)
+    urlhaus.urlhaus_session = mock_response_instance
+
+    response = urlhaus.make_urlhaus_request(to_lookup)
+
+    mock_response_instance.post.assert_called_once_with(
+        expected_url, to_lookup, headers=mock_response_instance.headers
+    )
+    assert response == mock_response
+
+
+@pytest.mark.parametrize(
+    "to_lookup",
+    [
+        ({"url": "https://example.com"}),
+    ],
+)
+@patch("modules.threat_intelligence.urlhaus.requests.session")
+def test_make_urlhaus_request_connection_error(
+    mock_response, mock_db, to_lookup
+):
+    """Test the ConnectionError handling in make_urlhaus_request."""
+    mock_response_instance = Mock()
+    mock_response.return_value = mock_response_instance
+    mock_response_instance.post.side_effect = (
+        requests.exceptions.ConnectionError
+    )
+
+    urlhaus = URLhaus(mock_db)
+    urlhaus.urlhaus_session = mock_response_instance
+    with patch.object(
+        urlhaus, "create_urlhaus_session"
+    ) as mock_create_session:
+        mock_create_session.side_effect = requests.exceptions.ConnectionError
+        with pytest.raises(requests.exceptions.ConnectionError):
+            urlhaus.make_urlhaus_request(to_lookup)
```

Comment:
what are you testing here? shouldn't we be checking that the return in None in case of a Connection error? also why did you add a side effect for create_urlhaus_session() ? isnt this `  mock_response_instance.post.side_effect ` enough?

Reply:
@AlyaGomaa  I am so sorry for this is incomplete i was trying something else here.